core/vm/evm.go

```diff
@@ -426,11 +426,17 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gasRemainin
 	ret, err = run(evm, contract, nil, false)
 
 	// EIP-170: Contract code size limit
-	if err == nil && evm.chainRules.IsSpuriousDragon && len(ret) > params.MaxCodeSize {
+	if err == nil && evm.chainRules.IsSpuriousDragon {
 		// Gnosis Chain prior to Shanghai didn't have EIP-170 enabled,
 		// but EIP-3860 (part of Shanghai) requires EIP-170.
 		if !evm.chainRules.IsAura || evm.config.HasEip3860(evm.chainRules) {
-			err = ErrMaxCodeSizeExceeded
+			if evm.chainConfig.Bor != nil && evm.chainConfig.Bor.IsAhmedabad(evm.Context.BlockNumber) {
+				if len(ret) > params.MaxCodeSizePostAhmedabad {
```

Comment:
there's a lot of unnecessary deep nesting that can be simplified if we set a `maxCodeSize` variable before the outmost if.

e.g.

```
var maxCodeSize int
if evm.chainConfig.Bor != nil && evm.chainConfig.Bor.IsAhmedabad(evm.Context.BlockNumber) {
    maxCodeSize = params.MaxCodeSizePostAhmedabad
} else {
    maxCodeSize = params.MaxCodeSizePostAhmedabad
}

if err == nil && evm.chainRules.IsSpuriousDragon && len(ret) > maxCodeSize {
    // same as before
}
```